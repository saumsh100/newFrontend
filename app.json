  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
