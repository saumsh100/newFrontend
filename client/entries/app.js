
import '../apps/default';
import '../apps/Dashboard';
