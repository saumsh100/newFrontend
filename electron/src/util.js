
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
