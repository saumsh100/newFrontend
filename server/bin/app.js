
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
