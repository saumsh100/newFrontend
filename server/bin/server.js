
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
