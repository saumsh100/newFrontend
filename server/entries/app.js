
import '../../client/apps/default';
import '../../client/apps/Dashboard';
