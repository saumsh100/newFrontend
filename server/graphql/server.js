
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

