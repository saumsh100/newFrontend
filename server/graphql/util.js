
import { getOffsetWithDefault, offsetToCursor } from 'graphql-relay';

/**
 * A simple function that accepts an array and connection arguments, and returns
 * a connection object for use in GraphQL
 *
 * This is a fork of connectionFromArraySlice
 * from https://github.com/graphql/graphql-relay-js/blob/master/src/connection/arrayconnection.js#L65
 *
 * Diferente form the original, is this case we build the cursor and pageInfo based on args
 * whitout slicing the array.
 * That way we can pass the already sliced or filtered array to the function.
 *
 * For now, this only works with forward pagination.
 */
// eslint-disable-next-line import/prefer-default-export
export function connectionFromArrayWithoutSlice(array, args, meta) {
  const { after, first, last } = args;
  const { arrayLength } = meta;

  if (typeof first === 'number') {
    if (first < 0) {
      throw new Error('Argument "first" must be a non-negative integer');
    }
  }
  if (typeof last === 'number') {
    if (last < 0) {
      throw new Error('Argument "last" must be a non-negative integer');
    }
  }

  const afterOffset = getOffsetWithDefault(after, 0);

  const startOffset = Math.max(afterOffset, -1) + 1;
  const endOffset = startOffset + first;

  const edges = array.map((value, index) => ({
    cursor: offsetToCursor(startOffset + index),
    node: value,
  }));

  const firstEdge = edges[0];
  const lastEdge = edges[edges.length - 1];
  const lowerBound = after ? afterOffset + 1 : 0;
  const upperBound = arrayLength;

  return {
    edges,
    pageInfo: {
      startCursor: firstEdge ? firstEdge.cursor : null,
      endCursor: lastEdge ? lastEdge.cursor : null,
      hasPreviousPage: typeof last === 'number' ? startOffset > lowerBound : false,
      hasNextPage: typeof first === 'number' ? endOffset < upperBound : false,
    },
  };
}
