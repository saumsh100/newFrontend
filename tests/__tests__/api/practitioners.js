
import request from 'supertest';
import app from '../../../server/bin/app';
import generateToken from '../../util/generateToken';
import { Practitioner } from '../../../server/models';
import wipeModel, { wipeAllModels } from '../../util/wipeModel';
import { accountId, seedTestUsers } from '../../util/seedTestUsers';
import { practitionerId, practitioner, seedTestPractitioners } from '../../util/seedTestPractitioners';
import { weeklySchedule, weeklyScheduleId, seedTestWeeklySchedules } from '../../util/seedTestWeeklySchedules';


describe('/api/practitioners', () => {
  // Seed with some standard user data
  let token = null;
  beforeAll(async () => {
    await seedTestUsers();
    token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
  });

  afterAll(async () => {
    await wipeAllModels();
  });

  describe('GET /', () => {
    beforeEach(async () => {
      await seedTestPractitioners();
    });

    test('get all practitioners', () => {
      return request(app)
        .get('/api/practitioners')
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(200)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });

    test('get all practitioners with weeklySchedule', () => {
      return request(app)
        .get('/api/practitioners?join=weeklySchedule')
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(200)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });

    test('/:practitionerId - get a practitioner', () => {
      return request(app)
        .get(`/api/practitioners/${practitionerId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('POST /', () => {
    beforeEach(async () => {
      //await seedTestPractitioners();
     // await seedTestWeeklySchedules();
      await wipeModel(Practitioner);
    });

    // TODO: Get help on this

    test('create practitioner', () => {
      return request(app)
        .post('/api/practitioners')
        .set('Authorization', `Bearer ${token}`)
        .send(Object.assign({},
          practitioner,{
            weeklyScheduleId,
          }
        ))
        .expect(201)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('PUT /', () => {
    beforeEach(async () => {
      await seedTestPractitioners();
    });

    test('/:practitionerId - update practitioner', () => {
      return request(app)
        .put(`/api/practitioners/${practitionerId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          firstName: 'Updated',
        })
        .expect(200)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });

    // TODO: Daily schemas have generated timestamps
    /*
    test.only('/:practitionerId/customSchedule - update weekly schedule', () => {
      const updatedWeeklySchedule = weeklySchedule;
      updatedWeeklySchedule.createdAt = '2017-07-19T00:18:30.932Z';
      return request(app)
        .put(`/api/practitioners/${practitionerId}/customSchedule`)
        .set('Authorization', `Bearer ${token}`)
        .send(Object.assign({ accountId }, weeklySchedule))
        .expect(201)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
          console.log(JSON.stringify(body));
        });
    });
    */

  });

  describe('DELETE /', () => {
    beforeEach(async () => {
      await seedTestPractitioners();
    });

    test('/:practitionerId - delete a practitioner', () => {
      return request(app)
        .delete(`/api/practitioners/${practitionerId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(204)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });
  });
});

