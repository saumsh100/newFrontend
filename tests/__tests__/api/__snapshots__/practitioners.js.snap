// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/practitioners DELETE / /:practitionerId - delete a practitioner 1`] = `Object {}`;

exports[`/api/practitioners GET / /:practitionerId - get a practitioner 1`] = `
Object {
  "entities": Object {
    "practitioners": Object {
      "497ff59a-4bae-4013-bdce-b6b5be91a1f5": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "firstName": "Colonel",
        "fullAvatarUrl": null,
        "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "isActive": true,
        "isCustomSchedule": false,
        "isHidden": false,
        "lastName": "Sanders",
        "weeklyScheduleId": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
      },
    },
  },
  "result": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
}
`;

exports[`/api/practitioners GET / get all practitioners 1`] = `
Object {
  "entities": Object {
    "practitioners": Object {
      "497ff59a-4bae-4013-bdce-b6b5be91a1f5": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "firstName": "Colonel",
        "fullAvatarUrl": null,
        "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "isActive": true,
        "isCustomSchedule": false,
        "isHidden": false,
        "lastName": "Sanders",
        "weeklyScheduleId": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
      },
    },
  },
  "result": Array [
    "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
  ],
}
`;

exports[`/api/practitioners GET / get all practitioners with weeklySchedule 1`] = `
Object {
  "entities": Object {
    "practitioners": Object {
      "497ff59a-4bae-4013-bdce-b6b5be91a1f5": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "firstName": "Colonel",
        "fullAvatarUrl": null,
        "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "isActive": true,
        "isCustomSchedule": false,
        "isHidden": false,
        "lastName": "Sanders",
        "weeklySchedule": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
        "weeklyScheduleId": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
      },
    },
    "weeklySchedules": Object {
      "8ce3ba61-60cd-40c6-bc85-c018cabd4a40": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "friday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "id": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
        "monday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "saturday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "sunday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "thursday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "tuesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "wednesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
      },
    },
  },
  "result": Array [
    "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
  ],
}
`;

exports[`/api/practitioners POST / create practitioner 1`] = `
Object {
  "entities": Object {
    "practitioners": Object {
      "497ff59a-4bae-4013-bdce-b6b5be91a1f5": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "firstName": "Colonel",
        "fullAvatarUrl": null,
        "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "isActive": true,
        "isCustomSchedule": false,
        "isHidden": false,
        "lastName": "Sanders",
        "weeklySchedule": "1f56957e-8766-4c75-992c-3a95069a9589",
        "weeklyScheduleId": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
      },
    },
    "weeklySchedules": Object {
      "1f56957e-8766-4c75-992c-3a95069a9589": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-24T20:12:50.306Z",
        "friday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "id": "1f56957e-8766-4c75-992c-3a95069a9589",
        "monday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "saturday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "sunday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "thursday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "tuesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "wednesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
      },
    },
  },
  "result": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
}
`;

exports[`/api/practitioners PUT / /:practitionerId - update practitioner 1`] = `
Object {
  "entities": Object {
    "practitioners": Object {
      "497ff59a-4bae-4013-bdce-b6b5be91a1f5": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "createdAt": "2017-07-19T00:14:30.932Z",
        "firstName": "Updated",
        "fullAvatarUrl": null,
        "id": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "isActive": true,
        "isCustomSchedule": false,
        "isHidden": false,
        "lastName": "Sanders",
        "services": Array [],
        "weeklyScheduleId": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
      },
    },
  },
  "result": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
}
`;
